### ISSUE TITLE
VERSION 0.7

### ISSUE THREAD

**Issue by alexcb** [earthly/earthly#2237]
This is a high catch-all ticket for the upcoming VERSION 0.7 release.

The following features will be enabled by default:
 - explicit-global
 - check-duplicate-images
 - earthly-version-arg
 - use-cache-command
 - use-host-command
 - use-copy-link
 - new-platform
 - no-tar-build-output
 - use-no-manifest-list
 - use-chmod
 - shell-out-anywhere
 - use-registry-for-with-docker
 - earthly-locally-arg
- use-project-secrets
- use-pipelines
- wait

The following feature(s) will remain as experimental, due to them (it) being too new:
- try

TODO:
 - [x] Draft release notes
 - [x] Finish up release notes (add information about all the breaking changes, expand on the meaning of each feature flag being flipped)
 - [x] Add docs for PIPELINE, TRIGGER, PROJECT, TRY (+FINALLY)
 - [ ] Update docs to point to 0.7 in all examples #2587 
 - [x] Update docs for commands that are being GA'd in 0.7 to no longer say "beta" or "experimental"
 - [x] Edit existing secrets-related docs to update to the new secrets (+ add migration information)
 - [x] Finish Podman support, including necessary docs
 - [ ] [Later] Update all syntax highlighting packages with the new commands: PIPELINE, TRIGGER, PROJECT, CACHE, WAIT, TRY, FINALLY.
 - [x] Remove `--save-inline-cache` and `--use-inline-cache` from `--ci`. They seem to be problematic in some cases and we don't know what the cause of this is (see e.g. https://github.com/earthly/earthly/issues/1187)
 - [x] Force using the VERSION command at the top of every Earthfile (specifying `VERSION 0.5` is still allowed, but it has to be explicit)
 - [x] #2544
 - [ ] [Later] #2178
 - [x] #1783
 - [x] #2272
 - [x] #2218
 - [x] #2609

Notes:
 - We will do one or more `v0.7-rc<n>` release candidates prior to the full v0.7.0 release

**Comment by alexcb** [comment]
#2269 contains flags for docs

**Comment by alexcb** [comment]
we will aim to release `WAIT` in VERSION 0.7 too; however need to fix #2218 first

**Comment by r-LaForge** [comment]
The change of behaviour that comes with the `--wait-block` flag will fix a segfault encountered with --push and SAVE ARTIFACT.

Closes https://github.com/earthly/earthly/issues/2286

**Comment by alexcb** [comment]
> The change of behaviour that comes with the `--wait-block` flag will fix a segfault encountered with --push and SAVE ARTIFACT.
> 
> Closes #2286

@r-LaForge We will still be supporting `VERSION 0.6` once `VERSION 0.7` is released (for at least a year, if not longer); ideally we'll want to fix the bug affecting 0.6 (however it'll be a lower priority since a work-around exists)

**Comment by rrjjvv** [comment]
> Force using the VERSION command at the top of every Earthfile

Would this be for the Earthfile being directly invoked?  Every Earthfile utilized in the build chain?  Or some subset (e.g. `IMPORT` vs. target vs. UDC, remote vs. local, etc.)?

**Comment by alexcb** [comment]
> > Force using the VERSION command at the top of every Earthfile
> 
> Would this be for the Earthfile being directly invoked? Every Earthfile utilized in the build chain? Or some subset (e.g. `IMPORT` vs. target vs. UDC, remote vs. local, etc.)?

We were thinking every `Earthfile`; however just initially forcing it on the directly called `Earthfile` might be a more gentle nudge -- since it makes importing remote UDCs (which one has no control over) impossible.

**Comment by rrjjvv** [comment]
> otherwise it makes importing remote UDCs (which one has no control over) impossible

Exactly why I asked the question (e.g. https://github.com/earthly/lib/issues/5, which is blocked by https://github.com/earthly/lib/issues/7) 😄.  Enforcing it for all Earthfiles referenced locally seems reasonable though (although there could still be edge cases like git submodules, where the Earthfile is 'local', yet still outside of your control... not sure how you can cleanly differentiate things like that).

**Comment by alexcb** [comment]
we will also implement `--pas-args` https://github.com/earthly/earthly/issues/1891

which will be needed as an alternative to https://github.com/earthly/earthly/pull/2451 which is required to fix errors such as:

```
Error: build target: build main: failed to solve: SAVE IMAGE testimg:latest was already declared (none had --platform values)
```

which now show up when `--wait` is enabled and the same target is built using different (unused) args.

These errors are blocking [a fix for](https://github.com/earthly/earthly/pull/2444) https://github.com/earthly/earthly/issues/2218

**Comment by alexcb** [comment]
`VERSION 0.7`

should be equivalent to:

```
VERSION                             \
--check-duplicate-images            \
--ci-arg                            \
--earthly-git-author-args           \
--earthly-locally-arg               \
--earthly-version-arg               \
--explicit-global                   \
--git-commit-author-timestamp       \
--new-platform                      \
--no-tar-build-output               \
--shell-out-anywhere                \
--use-cache-command                 \
--use-chmod                         \
--use-copy-link                     \
--use-host-command                  \
--use-no-manifest-list              \
--use-pipelines                     \
--use-project-secrets               \
--wait-block                        \
0.6
```

The above can be helpful for pinpointing which feature leads to incompatabilities
